### PR TITLE
dummy inputs to cpu & model to cpu before conversion

### PR DESCRIPTION
**Pull request by trajepl**
To avoid the case where user define the customized evaluation functions and forget to reset the device back to cpu.
Further fix for PR: https://github.com/microsoft/Olive/pull/177

### CODE REVIEW

File: olive/passes/pytorch/qat_utils.py [hunk 1]
```diff
@@ -168,7 +169,7 @@ def create_dummy_input(self):
         # dummy inputs
         dummy_inputs = []
         for input_shape, input_type in zip(self.config.input_shapes, input_types):
-            dummy_inputs.append(torch.zeros(input_shape, dtype=input_type))
+            dummy_inputs.append(torch.zeros(input_shape, dtype=input_type, device=Device.CPU))
```
Comment (guotuofeng) [inline olive/passes/pytorch/qat_utils.py:172]: better to postpone the data to device when it is used.

File: olive/passes/onnx/conversion.py [hunk 2]
```diff
@@ -119,6 +120,7 @@ def _initialize(self):
     def _run_for_config(self, model: PyTorchModel, config: Dict[str, Any], output_model_path: str) -> ONNXModel:
         pytorch_model = model.load_model()
         pytorch_model.eval()
+        pytorch_model.to(Device.CPU)
```
Comment (guotuofeng) [inline olive/passes/onnx/conversion.py:123]: better to add unit test

**Review by guotuofeng — CHANGES_REQUESTED** [review]
Please add unit test for the bug

File: olive/passes/pytorch/qat_utils.py (hunk 1 above)
Comment (trajepl) [reply olive/passes/pytorch/qat_utils.py]: Done.

File: olive/passes/onnx/conversion.py (hunk 2 above)
Comment (trajepl) [reply olive/passes/onnx/conversion.py]: Hard to directly test from unit tests. Maybe let us kick off the e2e tests for gpu in next pr when the gpu-cluster is ready.

**Comment by trajepl** [comment]
> Please add test for the bug
> 
> 1. model on cpu but data on gpu
> 2. model on gpu but data on cpu

Will raise it up to team to see what is the available gpu clusters for e2e test in later PR.

File: olive/passes/onnx/conversion.py [hunk 3]
```diff
@@ -119,6 +120,9 @@ def _initialize(self):
     def _run_for_config(self, model: PyTorchModel, config: Dict[str, Any], output_model_path: str) -> ONNXModel:
         pytorch_model = model.load_model()
         pytorch_model.eval()
+        # put pytorch_model and dummy_inputs at the same device
+        torch_device = pytorch_model.device
```
Comment (guotuofeng) [inline olive/passes/onnx/conversion.py:124]: seems the pipeline fail here since there is no device attribute on model?

**Comment by guotuofeng** [comment]
> Please add test for the bug
> 
> 1. model on cpu but data on gpu
> 2. model on gpu but data on cpu

Please add TODO for test if we'd like to add test in next PR.

File: olive/passes/onnx/conversion.py (hunk 3 above)
Comment (trajepl) [reply olive/passes/onnx/conversion.py:124]: https://stackoverflow.com/questions/58926054/how-to-get-the-device-type-of-a-pytorch-module-conveniently

Look like it is impossible to get device of model, I just put model and data to cpu direcly.

Comment (guotuofeng) [reply olive/passes/onnx/conversion.py:124]: sure

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**